docs/src/components/Pre.tsx

```diff
@@ -0,0 +1,97 @@
+import {Button, CopyToClipboard} from "nextra/components";
+import {WordWrapIcon} from "nextra/icons";
+import type {ComponentProps, ReactElement} from "react";
+import {useCallback, useRef} from "react";
+import {Javascript, Json, Typescript} from "./icons";
+
+function cn(...classes: Array<string | boolean | undefined>) {
+  return classes.filter(Boolean).join(" ");
+}
+
+export const Pre = ({
+  children,
+  className,
+  hasCopyCode,
+  filename,
+  ...props
+}: ComponentProps<"pre"> & {
+  filename?: string;
+  hasCopyCode?: boolean;
+}): ReactElement => {
+  const preRef = useRef<HTMLPreElement | null>(null);
+
+  const toggleWordWrap = useCallback(() => {
+    const htmlDataset = document.documentElement.dataset;
+    const hasWordWrap = "nextraWordWrap" in htmlDataset;
+    if (hasWordWrap) {
+      delete htmlDataset.nextraWordWrap;
+    } else {
+      htmlDataset.nextraWordWrap = "";
+    }
+  }, []);
+
+  const language = props["data-language" as keyof typeof props] as string;
+  const Icon = {
+    js: Javascript,
+    jsx: Javascript,
```

Comment:
really nitpicking now but maybe a React icon for jsx/tsx 🤔

Reply:
Hahaha you are totally fine. I could also add in an optional override prop so we can specify solid

Reply:
Agreed a react icon for this would be nice for now, can update solid to be specific in future PR